client/components/app.jsx

```diff
@@ -0,0 +1,136 @@
+import React, { Component } from 'react';
+import Header from './layout/Header';
+
+export default class TipCalculator extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      tipPercentage: {
+        '15%': 0.15,
+        '16%': 0.16,
+        '17%': 0.17,
+        '18%': 0.18,
+        '19%': 0.19,
+        '20%': 0.20,
+      },
+      tipSelected: null,
+      checkTotal: '',
+      checkTotalSet: false,
+      tipTotal: null,
+      grandTotal: null,
+      fateDecides: false,
+    };
+    this.onTotalCostChange = this.onTotalCostChange.bind(this);
+    this.onSetClick = this.onSetClick.bind(this);
+    this.calculateGrandTotal = this.calculateGrandtotal.bind(this);
+    this.handleTipPercentageClick = this.handleTipPercentageClick.bind(this);
+    this.handleReset = this.handleReset.bind(this);
+    this.letFateDecide = this.letFateDecide.bind(this);
+  }
+
+  onTotalCostChange(e) {
+    const name = e.target.name, value = e.target.value;
```

Comment:
```suggestion
    const name = e.target.name,
          value = e.target.value;
```
I always suggest keeping one thing per line for simple quick-scan readability.